### ISSUE TITLE
UnboundLocalError: local variable 'dispatch_tuple' referenced before assignment

### ISSUE THREAD

**Issue by Lcarey** [HadrienG/InSilicoSeq#69]
Hi,
ISS looks interesting. I'd like to simulate error rates from aligned reads.
I installed via pip3 (python 3.6.4) and tried to generate a model, but got an error: 

```
~/.local/bin/iss model --bam test2.bam --output test
INFO:iss.app:Starting iss model
INFO:iss.app:Using KDE ErrorModel
INFO:iss.bam:Reading bam file: test2.bam
Traceback (most recent call last):
  File "/home/lcarey/.local/bin/iss", line 11, in <module>
    sys.exit(main())
  File "/home/lcarey/.local/lib/python3.6/site-packages/iss/app.py", line 417, in main
    args.func(args)
  File "/home/lcarey/.local/lib/python3.6/site-packages/iss/app.py", line 214, in model_from_bam
    bam.to_model(args.bam, args.output)
  File "/home/lcarey/.local/lib/python3.6/site-packages/iss/bam.py", line 180, in to_model
    for pos, indel in modeller.dispatch_indels(read):
  File "/home/lcarey/.local/lib/python3.6/site-packages/iss/modeller.py", line 330, in dispatch_indels
    yield dispatch_tuple
UnboundLocalError: local variable 'dispatch_tuple' referenced before assignment
```

I've uploaded a bam file that fails, and a subset of that file that works: 
https://www.dropbox.com/sh/vl1zhz59u8nyela/AAAzAAn4fgdmzixe9cQDCa4Sa?dl=0

**Comment by Lcarey** [comment]
Hi Hadrien,

Cool. 

Did you think about learning context-dependent error profiles from the .bam file? My understanding of Illumina errors is that they also depend on the -1nt, and to a lesser extent on the -2nt. (eg https://www.ncbi.nlm.nih.gov/pubmed/18660515 and others). 

We're interested in using ultra-high coverage (~milllion x) coverage to identify subclonal mutations. 

-Lucas

**Comment by HadrienG** [comment]
Hi! 

Sorry for the delay. The bug you reported is solved with 1.2.1.
For answering your question, we're thinking to give HMMs a try, but I'm still struggling with how to best handle the dependance on the -2nt event. But hopefully there will be a HMM error profile option for version 2 😃
